troposphere/static/js/components/common/ui/CopyButton.jsx

```diff
@@ -0,0 +1,242 @@
+    //
+    // Why is it here? To ensure:
+    // 1. the element is able to have focus and selection.
+    // 2. if element was to flash render it has minimal visual impact.
+    // 3. less flakyness with selection and copying which **might** occur if
+    //    the textarea element is not visible.
+    //
+    // The likelihood is the element won't even render, not even a flash,
+    // so some of these are just precautions. However in IE the element
+    // is visible whilst the popup box asking the user for permission for
+    // the web page to copy to the clipboard.
+    //
+
+    // Place in top-left corner of screen regardless of scroll position.
+    textArea.style.position = 'fixed';
+    textArea.style.top = 0;
+    textArea.style.left = 0;
+
+    // Ensure it has a small width and height. Setting to 1px / 1em
+    // doesn't work as this gives a negative w/h on some browsers.
+    textArea.style.width = '2em';
+    textArea.style.height = '2em';
+
+    // We don't need padding, reducing the size if it does flash render.
+    textArea.style.padding = 0;
+
+    // Clean up any borders.
+    textArea.style.border = 'none';
+    textArea.style.outline = 'none';
+    textArea.style.boxShadow = 'none';
+
+    // Avoid flash of white box if rendered for any reason.
+    textArea.style.background = 'transparent';
+
+    textArea.value = text;
+
+    document.body.appendChild(textArea);
+
+    textArea.select();
+
+    try {
+        var successful = document.execCommand('copy');
+        var msg = successful ? 'successful' : 'unsuccessful';
+    } catch (err) {
+        reportException(err);
+    };
+
+    document.body.removeChild(textArea);
+};
+
+export default React.createClass({
+    getInitialState() {
+        return {
+            feedbackState: "WAIT",
+            canCopy: false,
+        }
+    },
+
+    componentDidMount() {
+        // Support for execCommand is good 
+        // http://caniuse.com/#search=document.execCommand
+
+        // The assumption is that if we have the execCommand method
+        // we also have "copy"
+
+        try {
+            var canCopy = document.execCommand;
+            this.setState({
+                canCopy,
+            });
+        } catch (err) {
+            reportException(err);
+            this.setState({
+                canCopy: false,
+            });
+        };
+    },
+
+    onClick() {
+        const { text } = this.props;
+        copyTextToClipboard(text);
+
+        this.setState({
+            feedbackState: "FLOAT"
+        }, () => {
+            setTimeout( 
+                () => { 
+                    this.setState({
+                        feedbackState: "FADEOUT"
+                }, () => {
+                    setTimeout(
+                    () => {    
+                        this.setState({
+                            feedbackState: "WAIT"   
+                        })
+                    } , 300)
+                })}, 300)
+            }
+        );
```

Comment:
This is some gross callback tangle. Once we refactor to JSS ( soon, I've got a branch started ) this can be more elegantly handled with CSS animations. Since this is contained in the component and works, shouldn't be a concern with this PR.